src/util/frame_limiter.rs

```diff
@@ -0,0 +1,123 @@
+//! Frame limiter
+
+use std::thread::{sleep, yield_now};
+use std::time::{Duration, Instant};
+
+/// Frame rate limiting strategy
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum FrameRateLimitStrategy {
+    /// No limit, spin as fast as possible
```

Comment:
Would like to add "WARNING: THIS MAY PRODUCE SYSTEM INSTABILITY DUE TO CPU STARVATION"

Reply:
Alternatively: If the strategy for this variant was to call `yield_now` it might not be as much of a concern.